app/src/main/java/com/kylecorry/trail_sense/navigation/ui/NavigatorFragment.kt

```diff
@@ -197,25 +195,50 @@ class NavigatorFragment : Fragment() {
             true
         }
 
-        binding.navigationOpenArCamera.setOnClickListener { view ->
+        binding.navigationOpenArCamera.setOnClickListener {
             if (!viewCameraBindToLifecycle) {
                 binding.viewCamera.bindToLifecycle(viewLifecycleOwner)
                 viewCameraBindToLifecycle = true
             }
             binding.viewCameraLine.visibility = if (binding.viewCamera.visibility == View.VISIBLE) View.INVISIBLE else View.VISIBLE
             binding.viewCamera.visibility = if (binding.viewCamera.visibility == View.VISIBLE) View.INVISIBLE else View.VISIBLE
+            if (userPrefs.navigation.rightQuickAction == QuickActionType.Flashlight) {
+                    binding.navigationRightQuickAction.isClickable = if (binding.viewCamera.visibility == View.VISIBLE) false else true
```

Comment:
FYI, there's a kotlin property called isVisible (you can get and set) that you can use instead of the inline if statements (binding.viewCamera.isVisible), that is a bit cleaner

Reply:
I just replaced it, should be cleaner now.